main.py

```diff
@@ -1,37 +1,91 @@
+from copy import deepcopy
+
 MIN_NUMBER = -2147483648
 MAX_NUMBER = 2147483647
+options = {
+    1: "Find the determinant of the matrix\n"
+}
 
 
 def get_input(msg, quantity, interval):
-    condition = False
-    while not condition:
+    # Get input from user: msg - message to show, quantity of the accepted parameters that are included in the interval
+    good_ans_condition = False
+    while not good_ans_condition:
         ans = list(input(msg).split())
-        condition = (len(ans) == quantity or quantity == -1)
+        good_ans_condition = (len(ans) == quantity or quantity == -1)  # -1 means unlimited quantity of the parameters
         try:
             for element in ans:
                 if not (int(element) in interval):
-                    condition = False
+                    good_ans_condition = False
                     break
         except:
-            condition = False
-        if condition:
+            good_ans_condition = False
+        if good_ans_condition:
             if quantity == 1:
                 ans = int(ans[0])
             return ans
         else:
             print("Something went wrong...")
 
 
-def init_matrix(lines):
-    matrix = []
-    columns = -1
-    for i in range(1,lines+1):
-        line = get_input("Line " + str(i) + ": ", columns, range(MIN_NUMBER, MAX_NUMBER))
-        if i==1:
-            columns = len(line)
-        matrix.append(line)
-    return matrix, lines, columns
+def get_option():  # Print all the options and get user's response
+    print("Options: \n", options)
+    return get_input("Enter: ", 1, range(1, len(options) + 1))
+
+
+class Matrix:
+    def __init__(self):
+        # Create the Matrix size n*m:
+        self.matrix, self.n, self.m = self.get_matrix()
+
+    def init_matrix(self, lines):  # Initialization the matrix from the keyboard
+        matrix = []
+        columns = -1
+        for i in range(1, lines + 1):
+            line = get_input("Line " + str(i) + ": ", columns, range(MIN_NUMBER, MAX_NUMBER + 1))
+            if i == 1:
+                columns = len(line)
+            matrix.append(line)
+        return matrix, lines, columns
+
+    def get_matrix(self):  # Find out the number of lines in the matrix n*m.
+        self.matrix, n, m = self.init_matrix(
+            get_input("Enter the number of lines of the matrix n (1-99): ", 1, range(1, 100)))
+        return self.matrix, n, m
+
+    def det(self, matrix, n, m):  # Count determinant
+        if n != m:
+            print("The determinant of the matrix can be found only in a square matrix.")
+            return None
+        elif n == 1:
+            return int(matrix[0][0])
```

Comment:
Minor: here and everywhere in the Matrix class:
If your matrix can contain float numbers, then cast to int is wrong operation
If your matrix must contain only int numbers, then this case is redundant. In this case if you're afraid the there will be float number - make the validation in Matrix constructor, ensure there that Matrix cannot be built if there is at least one float element

Reply:
Float numbers will be added later